### PR TITLE
FC-1266 Fix/index hardening

### PR DESCRIPTION
**Pull request by bplatz**
This changes the transactor's reliance on the query peer's db message process and instead directly handles managing the current ledger state when acting as a transactor.

### CODE REVIEW

File: src/fluree/db/ledger/indexing.clj [hunk 1]
```diff
@@ -398,53 +400,134 @@
...
+(defn merge-index-block
+  "Attempts to merge newly completed index job with block-map.
+  Optimistically the new index will be one block behind the block-map,
+  but if not we will pause momentarily to try to see if things catch up
+  before throwing and error."
+  [session indexed-db {:keys [block flakes] :as block-map}]
+  (go-try
+    (let [indexed-block (:block indexed-db)
+          current?      (= indexed-block block)]
+      (log/warn "-------- merge-index-block: " current? indexed-block)
```
Comment (cap10morgan) [inline src/fluree/db/ledger/indexing.clj:458]: Should this stay at `warn` level?

File: src/fluree/db/ledger/indexing.clj [hunk 2]
```diff
@@ -398,53 +400,134 @@
...
+      (if current?
+        (do
+          (session/release-indexing-lock! session indexed-block)
+          (assoc block-map :db-after indexed-db))
+        (let [_         (log/warn "^^^^^^ PAUSING  ^^^^ merge-index-block: " indexed-block block)
```
Comment (cap10morgan) [inline src/fluree/db/ledger/indexing.clj:463]: Is there a session ID of some sort we can output to correlate this with the previous log message even if other things get logged in between? Might not be much of an issue in practice.

File: src/fluree/db/ledger/indexing.clj [hunk 3]
```diff
@@ -398,53 +400,134 @@
...
+                                 "Pending new block: " block ", latest block on disk: " (:block latest-db) ".")
+                            {:status 500 :error :db/unexpected-error}))))))))
+
+
+(defn merge-new-index
+  ([session block-map reindexed-db]
+   (merge-new-index session block-map reindexed-db 0))
+  ([session {:keys [flakes block] :as block-map} reindexed-db retries]
+   (log/warn "MERGING NEW INDEX: " {:old (:db-after block-map)
+                                    :new reindexed-db})
```
Comment (cap10morgan) [inline src/fluree/db/ledger/indexing.clj:481]: Still want this in here? At `warn` level? With shoutiness? ;)

File: src/fluree/db/ledger/indexing.clj [hunk 4]
```diff
@@ -398,53 +400,134 @@
...
+                              "Pending new block: " block ", latest block on disk: " (:block reindexed-db) ".")
+                         {:status 500 :error :db/unexpected-error}))
 
          :else
-         (do
-           (log/warn "Indexing process failed to obtain index lock. Extremely Unusual."
-                     {:network network :db (pr-str db) :block block :indexing? (session/indexing? session)})
-           false))))))
+         (let [_ (log/warn "###### MERGING NEW INDEX WAIT - retries: " retries
+                           "indexed-block: " indexed-block "current-block: " block)
```
Comment (cap10morgan) [inline src/fluree/db/ledger/indexing.clj:501]: Another log message that you may or may not want to keep (at `warn` level and/or with shouty all caps)

File: src/fluree/db/peer/http_api.clj [hunk 5]
```diff
@@ -489,7 +491,8 @@
                           (try (Integer/parseInt timeout)
                                (catch Exception _ 60000))
                           60000)
-        [header body]   (<?? (action-handler action* system action-param auth-map ledger request-timeout))
+        opts            (assoc params :timeout request-timeout)
+        [header body] (<?? (action-handler action* system action-param auth-map ledger opts))
```
Comment (cap10morgan) [inline src/fluree/db/peer/http_api.clj:495]: Alignment is off here

**Review by cap10morgan — APPROVED** [review]
Just some very minor docs, formatting, & logging stuff to consider / fix. No blockers.

File: src/fluree/db/ledger/indexing.clj [hunk 6]
```diff
@@ -398,53 +400,134 @@
...
+          (session/clear-db! session)                       ;; clear db cache to force reload
+          (async/put! pc indexed-db))))
+    pc))
+
+
+(defn merge-index-block
+  "Attempts to merge newly completed index job with block-map.
+  Optimistically the new index will be one block behind the block-map,
+  but if not we will pause momentarily to try to see if things catch up
+  before throwing and error."
```
[the remark this answers is not in the document]
Comment (bplatz) [reply src/fluree/db/ledger/indexing.clj]: 👍

File: src/fluree/db/ledger/indexing.clj (hunk 2 above)
Comment (bplatz) [reply src/fluree/db/ledger/indexing.clj]: Fixed logging I left in by mistake.

File: src/fluree/db/ledger/indexing.clj (hunk 4 above)
Comment (bplatz) [reply src/fluree/db/ledger/indexing.clj]: Fixed logging I left in by mistake.

File: src/fluree/db/peer/http_api.clj (hunk 5 above)
Comment (bplatz) [reply src/fluree/db/peer/http_api.clj]: Fixed alignment

File: src/fluree/db/ledger/indexing.clj (hunk 3 above)
Comment (bplatz) [reply src/fluree/db/ledger/indexing.clj]: Fixed logging I left in by mistake.

File: src/fluree/db/ledger/indexing.clj (hunk 1 above)
Comment (bplatz) [reply src/fluree/db/ledger/indexing.clj]: Fixed logging I left in by mistake.

**Pull request merged**